File: app/src/main/java/com/yvrun/officeprocess/mvp/net/BaseObserver.java
package com.yvrun.officeprocess.mvp.net;


import com.yvrun.officeprocess.mvp.contract.TestContract;

import java.lang.ref.WeakReference;

import io.reactivex.Observer;
import io.reactivex.disposables.Disposable;

public abstract class BaseObserver<T> implements Observer<T> {


    public BaseObserver(TestContract.TestPresenterImp testPresenterImp, WeakReference<TestContract.ITestView> view) {

    }

    @Override
    public void onSubscribe(Disposable d) {

    }

    @Override
    public void onNext(T t) {

    }

    @Override
    public void onError(Throwable e) {

    }

    @Override
    public void onComplete() {

    }

    protected abstract void success(T t);
    protected abstract void error(Throwable e);
    protected abstract void addDisposable(Disposable d);
}

File: kotlinappother/src/main/java/com/chengzhen/kotlinappother/mvp/net/BaseObserver.kt
package com.chengzhen.kotlinappother.mvp.net


import com.chengzhen.kotlinappother.mvp.contract.TestContract

import java.lang.ref.WeakReference

import io.reactivex.Observer
import io.reactivex.disposables.Disposable

abstract class BaseObserver<T>(testPresenterImp: TestContract.TestPresenterImp, view: WeakReference<TestContract.ITestView>) : Observer<T> {

    override fun onSubscribe(d: Disposable) {

    }

    override fun onNext(t: T) {

    }

    override fun onError(e: Throwable) {

    }

    override fun onComplete() {

    }

    protected abstract fun success(t: T)
    protected abstract fun error(e: Throwable)
    protected abstract fun addDisposable(d: Disposable)
}
